src/main/java/com/example/onlinebookstore/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,20 @@
+package com.example.onlinebookstore.security;
+
+import com.example.onlinebookstore.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        return userRepository.findByEmail(username).orElseThrow(() ->
+                new RuntimeException("Can't find user by email"));
```

Comment:
Are you searching by email? What you provide user name? Also please add details to error message